### ISSUE TITLE
autodoc maybe should ignore return annotations on classes

### ISSUE THREAD

**Issue by sorcio** [sphinx-doc/sphinx#4969]
Subject: autodoc adds "→ None" to classes

### Problem
We noticed on Trio (https://github.com/python-trio/trio/issues/529) that some classes that have a return annotation on `__init__` are rendered by autodoc with "→ None".

Example:
http://trio.readthedocs.io/en/latest/reference-core.html#trio.Event
`class trio.Event → None`

This class uses `attrs`, which [adds a `-> None` annotation to the generated `__init__` method](https://github.com/python-attrs/attrs/commit/fc033b41a11c9ef3e5fd7bee7f936427d4bb5f3d#diff-21756223332b04173cec8910fda6655bR1264).

A minimal example could be:

```python
class Foo:
    """Class docstring"""
    def __init__(self) -> None:
        pass
```

And `.. autoclass:: Foo` will generate `class Foo → None`.

I would expect autodoc to ignore the return annotation for the `__init__` method when it is used to build the signature for the class itself. A return annotation of `None` is legal for `__init__` but it doesn't make sense to pick it up to document the class signature.

**Comment by tk0miya** [comment]
+1: Absolutely!

**Comment by tk0miya** [comment]
Fixed by #4972.
Thank you for reporting
